src/main/java/network/brightspots/rcv/AuditableFile.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * RCTab
+ * Copyright (c) 2017-2023 Bright Spots Developers.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Purpose: Create a file that, on close, is read-only and has its hash added to the audit log.
+ * Design: Overrides the File class
+ * Conditions: Always.
+ * Version history: see https://github.com/BrightSpots/rcv.
+ */
+
+package network.brightspots.rcv;
+
+import java.io.File;
+
+final class AuditableFile extends File {
+  public AuditableFile(String pathname) {
+    super(pathname);
+  }
+
+  public void finalizeAndHash() {
+    boolean readOnlySucceeded = setReadOnly();
+    if (!readOnlySucceeded) {
+      Logger.warning("Failed to set file to read-only: %s", getAbsolutePath());
```

Comment:
Any chance that we want this type of error to trigger `severe` instead of `warning`?

Reply:
I opted for warning since I think there are valid cases where this will fail, though I'm not sure of them. For example, if saving to a mounted Google Drive folder, are you allowed to set a file to readonly?

I haven't tested any of this, but given that the function returns a boolean rather than throwing an exception, I assumed there would be valid, non-severe cases where this fails.